index.ts

```diff
@@ -1,16 +1,63 @@
 import { clean, inc, valid } from "https://deno.land/x/semver@v1.4.0/mod.ts";
+import { resolve } from "https://deno.land/std@0.204.0/path/mod.ts";
 
 import UserError from "./user-error.ts";
 import { checkPrerequisites, commitAndTag, GitError } from "./git.ts";
 
-const fileName = "VERSION";
+const textFileName = "VERSION";
+const codeFileName = "VERSION.ts";
+
+const codeRegex = /export[\s\S]+?const[\s\S]+?VERSION\s+=\s+"(.*?)";/g;
```

Comment:
Just thinking out loud here; this feels like a bit of a strange middle ground; it relies on the file contents being _pretty much_ a certain format, but not exactly (e.g. allows multiple spaces).

Alternatives I could see:
- Read this programmatically rather than using a regex? e.g. something like `eval`... Maybe that's a bad idea and opens up more problems than it solves.
- Make this _more_ strict; since we prescribe exactly how the file is written, we could prescribe exactly the formatting of it too; remove all variability and replace the `[\s\S]+?` and `\s+` with just a literal ` ` space.

What do you think?

(also, a minor nitpick, i'm not sure I understand the `\S` non-space matches, as this means something like this is currently valid: `exporting a very cool constant called VERSION = "2.0.0";`)

Reply:
Sorry for not really getting back to you on this one! Have been quite busy 😅 

I totally get where you're coming from and I agree - this isn't ideal. When I first published the PR I actually did perform an `import()` to get the value which is prettier... but runs into permission problems when run from remote. I don't know if `eval` would have the same problem, but either way it won't support proper TypeScript, not to mention the lack of support for `export` in `eval`.

To be fair though, this solution doesn't exactly support TypeScript features either, so I am leaning towards doing as you say, lock down the regex and expect the file to be a certain way. Although that comes with the problem that code formatters might automatically format the file and then we're back to square one again 😕 

So maybe iterating on the regex a little bit is the best option? Unless we want to pull in a proper parser that is of course 😂

Reply:
That's a good point about code formatters though. My main "blocking" request is to get rid of the `/S` matches as mentioned above, unless they're doing something I'm not seeing.